models/aws_project.rb

```diff
@@ -512,7 +512,8 @@ def get_data_out(date=DEFAULT_DATE)
   end
 
   def get_aws_instance_info
-    regions = InstanceLog.where(host: "AWS").select(:region).distinct.pluck(:region).sort
+    regions = AwsProject.where(host: "aws").map { |project| project.regions }.flatten.uniq | ["eu-west-2"]
```

Comment:
Having trouble testing this bit, is the `where(host: 'aws')` clause here redundant since we're already querying `AwsProjects`? Also, the `map` block can be shorthanded to `map(&:regions)`.

Reply:
One way to test it would be to remove `eu-west-2` from all of your AWS Projects' regions, then run this and see if `eu-west-2` is still included.

The `where` is not redundant, as both `AwsProjects` and `AzureProjects` are stored in the `Projects` table. But thinking about it, we can/ probably should make it redundant using some default scopes (which I'll add shortly).

Reply:
`AwsProject` and `AzureProject` are indeed both children of `Project`, but `Project` isn't being queried here; `AwsProject` is.

Reply:
Yes but they are both represented in one `projects` SQL table. We haven't defined any default scope, so unless specified, active record queries for `AwsProject` or `AzureProject` will just use all rows in the projects table. E.g. `AzureProject.last` will use the same table row as `Project.last`, as will `AwsProject.last` (regardless of its host, or any other attributes). 

![Screenshot from 2020-12-16 15-10-19](https://user-images.githubusercontent.com/59840834/102366664-d416f380-3fb0-11eb-90c5-b4c20b40122a.png)